src/components/Calendar.tsx

```diff
@@ -0,0 +1,166 @@
+                                <Msg id={ day }/>
+                            </Heading>
+                            <Text data-testid={ day }>
+                                { new Date(new Date(monday).setDate(monday.getDate() + index)).getDate() }
+                            </Text>
+                        </div>
+                    )) }
+                </div>
+                <div ref={ calendar } style={{
+                    alignItems: 'center',
+                    display:'flex',
+                    gap:'0.5rem',
+                    height: '100rem',
+                    justifyContent: 'start',
+                    width: '100%',
+                }}>
+                    { week.map((day, index) => (
+                        <div key={ day } style={{
+                            display: 'flex',
+                            flexDirection: 'column',
+                            gap: '1rem',
+                            height: '100%',
+                            justifyContent: 'space-between',
+                            padding: '0',
+                            width: '100%',
+                        }}>
+                            <ul data-testid={ `${day}-events` } style={{
+                                background:'lightgray',
+                                height: '100%',
+                                listStyle: 'none',
+                                margin: 0,
+                                padding:0 ,
+                                position: 'relative',
+                                width: '100%',
+                            }}>
+                                { getEventsOfTheDay(index + 1)?.map(event => (
+                                    <CalendarEvent
+                                        key={ event.id }
+                                        endTime={ event.end_time }
+                                        id={ event.id }
+                                        startTime={ event.start_time }
+                                    />
+                                )) }
+                            </ul>
+                        </div>
+                    )) }
+                </div>
+            </div>
+        </>
+    );
+};
+
+interface CalendarEventProps {
+    startTime: string;
+    endTime: string;
+    id: number;
+}
+
+const CalendarEvent = ({ startTime, endTime, id }: CalendarEventProps): JSX.Element => {
+
+    const getPos = (start: string, end: string) => {
+        const oneMinute = 100 / 1440;
+        const startTime = new Date(start);
+        const endTime = new Date(end);
+        const startFromMidnight = (startTime.getTime() - startTime.setHours(0, 0, 0, 0)) / 60000;
+        const endFromMidnight = (endTime.getTime() - endTime.setHours(0, 0, 0, 0)) / 60000;
+        const diff = endFromMidnight - startFromMidnight;
+
+        return {
+            height: `${diff * oneMinute}%`,
+            top: `${startFromMidnight * oneMinute}%`,
```

Comment:
Nice with the percentages! For some reason I had imagined using absolute pixels, but this is much better!